src/ext/VSExtension/ca/vsca.cpp

```diff
@@ -0,0 +1,360 @@
+
+        qwVersion = FILEMAKEVERSION(15, 0, 0, 0);
+        hr = S_OK;
+    }
+
+    hr = pConfiguration->EnumInstances(&pEnumInstances);
+    ExitOnFailure(hr, "Failed to get instance enumerator.");
+
+    do
+    {
+        hr = pEnumInstances->Next(1, rgpInstances, &cInstancesFetched);
+        if (SUCCEEDED(hr) && cInstancesFetched)
+        {
+            pInstance = rgpInstances[0];
+            if (pInstance)
+            {
+                if (pHelper)
+                {
+                    hr = pInstance->GetInstallationVersion(&bstrVersion);
+                    ExitOnFailure(hr, "Failed to get installation version.");
+
+                    hr = pHelper->ParseVersion(bstrVersion, &qwVersion);
+                    ExitOnFailure(hr, "Failed to parse installation version.");
+                }
+
+                for (DWORD i = 0; i < countof(vrgInstances); ++i)
+                {
+                    const VS_INSTANCE* pElem = &vrgInstances[0];
+
+                    if (pElem->qwMinVersion <= qwVersion && qwVersion <= pElem->qwMaxVersion)
+                    {
+                        hr = pElem->pfnProcessInstance(pInstance, qwVersion, FALSE);
+                        ExitOnFailure(hr, "Failed to process instance.");
+                    }
+                }
+            }
+
+            ReleaseNullObject(pInstance);
+        }
+    } while (SUCCEEDED(hr) && cInstancesFetched);
+
+    // Complete all registered processing functions.
+    for (DWORD i = 0; i < countof(vrgInstances); ++i)
+    {
+        const VS_INSTANCE* pElem = &vrgInstances[0];
+
+        if (pElem->qwMinVersion <= qwVersion && qwVersion <= pElem->qwMaxVersion)
+        {
+            hr = pElem->pfnProcessInstance(NULL, 0, TRUE);
+            ExitOnFailure(hr, "Failed to process latest instance.");
+        }
+    }
+
+LExit:
+    ReleaseBSTR(bstrVersion);
+    ReleaseObject(pInstance);
+    ReleaseObject(pEnumInstances);
+    ReleaseObject(pHelper);
+    ReleaseObject(pConfiguration);
+
+    if (fComInitialized)
+    {
+        ::CoUninitialize();
+    }
+
+    if (FAILED(hr))
+    {
+        er = ERROR_INSTALL_FAILURE;
+    }
+
+    return WcaFinalize(er);
+}
+
+static HRESULT InstanceIsGreater(
+    __in_opt ISetupInstance* pPreviousInstance,
+    __in DWORD64 qwPreviousVersion,
+    __in ISetupInstance* pCurrentInstance,
+    __in DWORD64 qwCurrentVersion
+    )
+{
+    HRESULT hr = S_OK;
+    FILETIME ftPreviousInstance = {};
+    FILETIME ftCurrentInstance = {};
+
+    if (qwPreviousVersion != qwCurrentVersion)
+    {
+        return qwPreviousVersion < qwCurrentVersion ? S_OK : S_FALSE;
+    }
+
+    hr = pPreviousInstance->GetInstallDate(&ftPreviousInstance);
+    ExitOnFailure(hr, "Failed to get previous install date.");
+
+    hr = pCurrentInstance->GetInstallDate(&ftCurrentInstance);
+    ExitOnFailure(hr, "Failed to get current install date.");
+
+    return 0 <= ::CompareFileTime(&ftPreviousInstance, &ftCurrentInstance) ? S_OK : S_FALSE;
```

Comment:
Technically, `=` should never happen with install dates since we block concurrent installs but I didn't want to specifically exclude the possibility, so figured if they are equal use the "current" instance.